robot/client/client.go

```diff
@@ -338,7 +420,7 @@ func (rc *RobotClient) Close(ctx context.Context) error {
 
 func (rc *RobotClient) checkConnected() error {
```

Comment:
do we need the checkConnected check in the other methods in the robot client at this point?

Reply:
we might not - let me see if tests still pass after I remove those extra checks.

Reply:
our unit tests fail, so we still have a subtle dependence on these checks - let's make this a follow-up

Reply:
sgtm